pages/index.tsx

```diff
@@ -123,6 +48,19 @@ function StartPage({ municipalities }: PropsType) {
   const [selectedDataset, setSelectedDataset] = useState<DatasetKey>(defaultDataset)
   const [selectedDataView, setSelectedDataView] = useState(normalizedDataView)
 
+  const [showCompanyData, setShowCompanyData] = useState(true)
+
+  const handleToggle = () => {
+    setShowCompanyData(!showCompanyData)
+    setSelectedDataset(defaultDataset)
+    setSelectedDataView(defaultDataView)
+
+    const dataGroup = showCompanyData ? 'geografiskt' : 'foretag'
+    const path = `/${dataGroup}/${defaultDataset}/${defaultDataView}`
+
+    router.push(path, undefined, { shallow: true })
+  }
+
   useEffect(() => {
     if (normalizedRouteDataset && isValidDataset(normalizedRouteDataset)) {
       setSelectedDataset(normalizedRouteDataset)
```

Comment:
Glitch: For municipalities, Server side state is initialized to show the defaultDataset. But then in this useEffect, we update based on the URL.

It would be much better to show the correct dataset based on the URL directly, since this would speed up rendering and prevent the view from changing unnecessarily.

Reply:
This will be fixed in #544